proposals/0010-vk-buffer-ref.md

```diff
@@ -51,7 +51,19 @@ The goal of this proposal is to have a solution that meets the following require
 
 ## Proposed solution
 
-Our solution is to add a new builtin type in the vk namespace that is a pointer to a buffer of a given type, `vk::BufferPointer<T,A>`. The template argument `T` must be a struct. `A` must be an integer and is the alignment in bytes of the pointer. If `A` is not specified, the alignment is assumed to be 16 bytes.
+Our solution is to add a new builtin type in the vk namespace that is a pointer to a buffer of a given type:
+
+```c++
+template <struct S, int align>
+class vk::BufferPointer {
+    vk::BufferPointer(const vk::BufferPointer&);
+    vk::BufferPointer& operator=(const vk::BufferPointer&);
+    vk::BufferPointer(const uint64_t);
+    S& Get() const;
+}
+```
+
+This class represents a pointer to a buffer of type struct `S`. `align` is the alignment in bytes of the pointer. If `align` is not specified, the alignment is assumed to be 16 bytes.
```

Comment:
Is the default alignment `16` or `max(16, alignof(S))`? Can we add that as the default value for the template parameter above?

Reply:
Can a struct have an alignment greater than 16? I had assumed not, hence the default of 16.

Reply:
I don't know that we enforce larger than 16-byte alignments on data types in HLSL right now, but certainly in C++ there are data types that require more restrictive alignments and there's no language-defined limit on alignment in HLSL.

For some C++ examples see: https://godbolt.org/z/YPh3Pfvr9

Reply:
OK, will use max formula.

Reply:
On further consideration, I am thinking that the default alignment should just be alignof(S). If you disagree, let me know.